crates/bin/pd/src/testnet/join.rs

```diff
@@ -304,8 +310,12 @@ pub async fn unpack_state_archive(archive_url: Url, output_dir: PathBuf) -> anyh
     }
 
     tracing::info!("archived node state unpacked to {}", pd_home.display());
-    // Post-extraction, clean up the downloaded tarball.
-    std::fs::remove_file(archive_filepath)?;
+
+    if !leave_archive {
+        // Post-extraction, clean up the downloaded tarball.
+        std::fs::remove_file(archive_filepath)?;
+    }
```

Comment:
it would be nice to complement this with an `else { tracing::info!(?path = archive_filepath, "leaving downloaded archive on disk"); }` branch to point users at the location of the tarball, for the sake of facilitating copy-pasting and saving someone the trouble of going looking for it.

Reply:
Great suggestion, I tacked this on just now after testing locally.